pkg/trtl/bench_test.go

```diff
@@ -0,0 +1,347 @@
+	if bench.trtl, err = trtl.New(bench.conf); err != nil {
+		return nil, err
+	}
+
+	// Create a bufconn listener(s) so that there are no actual network requests
+	bench.grpc = bufconn.New(benchbuf)
+
+	// Run the test server without signals, background routines or maintenance mode checks
+	go bench.trtl.Run(bench.grpc.Listener)
+
+	return bench, err
+}
+
+// cleanup the current temporary directory, configuration, and running services.
+func cleanup(bench *trtlBench) (err error) {
+	// Shutdown the trtl server if it is running
+	// This should shutdown all the running services and close the database
+	// Note that Shutdown should be graceful and not shutdown anything not running.
+	if bench.trtl != nil {
+		if err = bench.trtl.Shutdown(); err != nil {
+			return err
+		}
+	}
+
+	// Shutdown the gRPC connection if it's running
+	if bench.grpc != nil {
+		bench.grpc.Release()
+	}
+
+	// Cleanup the benchdb and delete any stray files
+	if bench.tmpdb != "" {
+		os.RemoveAll(bench.tmpdb)
+	}
+
+	return nil
+}
+
+func BenchmarkTrtlGet(b *testing.B) {
+	// Run setupTrtl
+	bench, e := setupTrtl(b)
+	require.NotNil(b, bench)
+	require.NoError(b, e)
+
+	// Start the gRPC client.
+	require.NoError(b, bench.grpc.Connect())
+	defer bench.grpc.Close()
+	cc := pb.NewTrtlClient(bench.grpc.Conn)
+	ctx := context.Background()
+
+	// Manually add some fixtures
+	// Create a put request
+	key := []byte("terminator")
+	catchphrase := []byte("hasta la vista, baby")
+	_, err := cc.Put(ctx, &pb.PutRequest{
+		Key:   key,
+		Value: catchphrase,
+	})
+	require.NoError(b, err)
+
+	// Run the trtl Get on a loop
+	b.ResetTimer()
+	var gErr error
+	for i := 0; i < b.N; i++ {
+		_, gErr = cc.Get(ctx, &pb.GetRequest{Key: key})
+	}
+	require.NoError(b, gErr)
+
+	// Run cleanup function
+	cleanup(bench)
+}
+
+func BenchmarkTrtlPut(b *testing.B) {
+	// Run setupTrtl
+	bench, e := setupTrtl(b)
+	require.NotNil(b, bench)
+	require.NoError(b, e)
+
+	// Start the gRPC client.
+	require.NoError(b, bench.grpc.Connect())
+	defer bench.grpc.Close()
+	cc := pb.NewTrtlClient(bench.grpc.Conn)
+	ctx := context.Background()
+
+	// Manually create some fixtures
+	key := []byte("sarah")
+	catchphrase := []byte("How's the knee?")
+
+	// Run the trtl Get on a loop
+	b.ResetTimer()
+	var err error
+	for i := 0; i < b.N; i++ {
+		_, err = cc.Put(ctx, &pb.PutRequest{
+			Key:   key,
+			Value: catchphrase,
+		})
+	}
+	require.NoError(b, err)
+
+	// Run cleanup function
+	cleanup(bench)
+}
+
+func BenchmarkTrtlDelete(b *testing.B) {
+	// Run setupTrtl
+	bench, e := setupTrtl(b)
+	require.NotNil(b, bench)
+	require.NoError(b, e)
+
+	// Start the gRPC client.
+	require.NoError(b, bench.grpc.Connect())
+	defer bench.grpc.Close()
+	cc := pb.NewTrtlClient(bench.grpc.Conn)
+	ctx := context.Background()
+
+	// Manually add some fixtures
+	var err error
+	for i := 0; i < b.N; i++ {
+		_, err = cc.Put(ctx, &pb.PutRequest{
+			Key:   []byte(fmt.Sprintf("t%d", i)),
+			Value: []byte(strconv.Itoa(i)),
+		})
+	}
```

Comment:
I'm not 100% sure this will work - I think that `b.N` will change during the benchmark depending on how long the op takes. I recommend using stop and start timer in the benchmark loop as follows:

```go
	// Reset the timer to focus only on the Delete
	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		// Stop timer and put key/value to delete 
		b.StopTimer()
		_, err = cc.Put(ctx, &pb.PutRequest{
			Key:   []byte(fmt.Sprintf("t%d", i)),
			Value: []byte(strconv.Itoa(i)),
		})
		require.NoError(b, err)
		
		// Start timer and perform the Delete 
		b.StartTimer()
                 _, err = cc.Delete(ctx, &pb.DeleteRequest{
			Key: []byte(fmt.Sprintf("t%d", i)),
		})		
	}
```